libs/sdk-backend-tiger/src/convertors/fromBackend/AccessControlConverter.ts

```diff
@@ -0,0 +1,56 @@
+// (C) 2022-2023 GoodData Corporation
+
+import {
+    idRef,
+    AccessGranularPermission,
+    IAvailableUserAccessGrantee,
+    IAvailableUserGroupAccessGrantee,
+    IGranularUserAccess,
+    IGranularUserGroupAccess,
+} from "@gooddata/sdk-model";
+import {
+    GrantedPermission,
+    UserAssignee,
+    UserGroupAssignee,
+    UserGroupPermission,
+    UserPermission,
+} from "@gooddata/api-client-tiger";
+
+const getPermissionLevels = (permissions: GrantedPermission[] = [], source: "direct" | "indirect") => {
+    return permissions
+        .filter((permission) => permission.source === source)
+        .map((permission) => permission.level as AccessGranularPermission);
+};
+
+export const convertUserPermission = (user: UserPermission): IGranularUserAccess => ({
+    type: "user",
+    user: {
+        ref: idRef(user.id),
+        uri: user.id,
+        email: user.email ?? user.id,
+        login: user.email ?? user.id,
+        fullName: user.name ?? user.id,
+    },
+    permissions: getPermissionLevels(user.permissions, "direct"),
+    inheritedPermissions: getPermissionLevels(user.permissions, "indirect"),
+});
+
+export const convertUserGroupPermission = (group: UserGroupPermission): IGranularUserGroupAccess => ({
+    type: "group",
+    userGroup: { ref: idRef(group.id), name: group.name ?? group.id },
+    permissions: getPermissionLevels(group.permissions, "direct"),
+    inheritedPermissions: getPermissionLevels(group.permissions, "indirect"),
+});
+
+export const convertUserAssignee = (user: UserAssignee): IAvailableUserAccessGrantee => ({
+    type: "user",
+    ref: idRef(user.id),
+    name: user.name ?? user.id,
```

Comment:
"email is used if the user's name and surname is missing"

Reply:
good catch, thank you